Stores/src/standard/java/com.infomaniak.lib.stores/InAppUpdateManager.kt

```diff
@@ -0,0 +1,150 @@
+        localSettings.isUserWantingUpdates = isUserWantingUpdate
+        onUserChoice(isUserWantingUpdate)
+    }
+
+    private val viewModel: StoresViewModel by lazy { ViewModelProvider(activity)[StoresViewModel::class.java] }
+
+    private val onUpdateDownloaded = { viewModel.toggleAppUpdateStatus(localSettings, isUpdateDownloaded = true) }
+    private val onUpdateInstalled = { viewModel.toggleAppUpdateStatus(localSettings, isUpdateDownloaded = false) }
+
+    // Create a listener to track request state updates.
+    private val installStateUpdatedListener by lazy {
+        InstallStateUpdatedListener { state ->
+            when (state.installStatus()) {
+                InstallStatus.DOWNLOADED -> {
+                    SentryLog.d(StoreUtils.APP_UPDATE_TAG, "OnUpdateDownloaded triggered by InstallStateUpdated listener")
+                    onUpdateDownloaded()
+                }
+                InstallStatus.INSTALLED -> {
+                    SentryLog.d(StoreUtils.APP_UPDATE_TAG, "OnUpdateInstalled triggered by InstallStateUpdated listener")
+                    onUpdateInstalled()
+                    unregisterAppUpdateListener()
+                }
+                else -> Unit
+            }
+        }
+    }
+
+    override fun onCreate(owner: LifecycleOwner) {
+        super.onCreate(owner)
+        observeAppUpdateDownload()
+    }
+
+    override fun onResume(owner: LifecycleOwner) {
+        super.onResume(owner)
+
+        viewModel.canInstallUpdate.value = localSettings.hasAppUpdateDownloaded
+        checkStalledUpdate()
+    }
+
+    override fun onStop(owner: LifecycleOwner) {
+        unregisterAppUpdateListener()
+        super.onStop(owner)
+    }
+
+    override fun checkUpdateIsAvailable() {
+        SentryLog.d(StoreUtils.APP_UPDATE_TAG, "Checking for update on GPlay")
+        appUpdateManager.appUpdateInfo.addOnSuccessListener { appUpdateInfo ->
+            SentryLog.d(StoreUtils.APP_UPDATE_TAG, "checking success")
+            if (appUpdateInfo.updateAvailability() == UpdateAvailability.UPDATE_AVAILABLE
+                && appUpdateInfo.isUpdateTypeAllowed(StoreUtils.UPDATE_TYPE)
+            ) {
+                SentryLog.d(StoreUtils.APP_UPDATE_TAG, "Update available on GPlay")
+                startUpdateFlow(appUpdateInfo, inAppUpdateResultLauncher)
+            }
+        }
+    }
+
+    override fun installDownloadedUpdate() {
+        localSettings.hasAppUpdateDownloaded = false
+        viewModel.canInstallUpdate.value = false
+        onInstallStart()
+        appUpdateManager.completeUpdate()
+            .addOnSuccessListener { onInstallSuccess?.invoke() }
+            .addOnFailureListener {
+                localSettings.resetUpdateSettings()
+                onInstallFailure(it)
+            }
+    }
+
+    private fun observeAppUpdateDownload() {
+        viewModel.canInstallUpdate.observe(activity) { isUploadDownloaded -> onInAppUpdateUiChange?.invoke(isUploadDownloaded) }
```

Comment:
`Upload` ? Plutot `Update` non?

Reply:
Good catch I thought I had renamed it, I must have missed this one